diff --git a/src/controller/src/rocprofvis_controller_graph.cpp b/src/controller/src/rocprofvis_controller_graph.cpp
index c7d76c1..14afb4e 100644
--- a/src/controller/src/rocprofvis_controller_graph.cpp
+++ b/src/controller/src/rocprofvis_controller_graph.cpp
@@ -432,36 +432,42 @@ rocprofvis_result_t Graph::GenerateLOD(uint32_t lod_to_generate, double start, d
 
                 start = std::max(start, min_ts);
                 end = std::min(end, max_ts);
-                ROCPROFVIS_ASSERT(start >= min_ts);
-                ROCPROFVIS_ASSERT(end <= max_ts);
-                double fetch_start = min_ts + (floor((start - min_ts) / segment_duration) * segment_duration);
-                double fetch_end = std::min(min_ts + (ceil((end - min_ts) / segment_duration) * segment_duration), max_ts);
-
-                GraphLODArgs args;
-                args.m_valid_range = range;
-                args.m_index       = 0;
-                result             = m_track->FetchSegments(
-                    fetch_start, fetch_end, &args,
-                    [](double start, double end, Segment& segment,
-                       void* user_ptr) -> rocprofvis_result_t {
-                        GraphLODArgs*       pair   = (GraphLODArgs*) user_ptr;
-                        rocprofvis_result_t result = kRocProfVisResultSuccess;
-
-                        if(pair->m_valid_range.first > segment.GetStartTimestamp() ||
-                           pair->m_valid_range.second < segment.GetMaxTimestamp())
-                        {
-                            result =
-                                segment.Fetch(start, end, pair->m_entries, pair->m_index);
-                            ROCPROFVIS_ASSERT(result == kRocProfVisResultSuccess);
-                        }
-                        return result;
-                    });
-
-                if(result == kRocProfVisResultSuccess)
+                if(end > start)
                 {
-                    result = GenerateLOD(lod_to_generate, start, end, args.m_entries);
-                    ROCPROFVIS_ASSERT(result == kRocProfVisResultSuccess);
-                    it->second.SetValidRange(start, end);
+                    double fetch_start =
+                        min_ts +
+                        (floor((start - min_ts) / segment_duration) * segment_duration);
+                    double fetch_end =
+                        std::min(min_ts + (ceil((end - min_ts) / segment_duration) *
+                                           segment_duration),
+                                 max_ts);
+
+                    GraphLODArgs args;
+                    args.m_valid_range = range;
+                    args.m_index       = 0;
+                    result             = m_track->FetchSegments(
+                        fetch_start, fetch_end, &args,
+                        [](double start, double end, Segment& segment,
+                           void* user_ptr) -> rocprofvis_result_t {
+                            GraphLODArgs*       pair   = (GraphLODArgs*) user_ptr;
+                            rocprofvis_result_t result = kRocProfVisResultSuccess;
+
+                            if(pair->m_valid_range.first > segment.GetStartTimestamp() ||
+                               pair->m_valid_range.second < segment.GetMaxTimestamp())
+                            {
+                                result = segment.Fetch(start, end, pair->m_entries,
+                                                                   pair->m_index);
+                                ROCPROFVIS_ASSERT(result == kRocProfVisResultSuccess);
+                            }
+                            return result;
+                        });
+
+                    if(result == kRocProfVisResultSuccess)
+                    {
+                        result = GenerateLOD(lod_to_generate, start, end, args.m_entries);
+                        ROCPROFVIS_ASSERT(result == kRocProfVisResultSuccess);
+                        it->second.SetValidRange(start, end);
+                    }
                 }
             }
         }
